src/smart-components/workflow/add-workflow-modal.js

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { withRouter } from 'react-router-dom';
+import { Modal } from '@patternfly/react-core';
+import { addNotification } from '@red-hat-insights/insights-frontend-components/components/Notifications';
+import { formFieldsMapper } from '@data-driven-forms/pf4-component-mapper';
+import FormRenderer from '../common/form-renderer';
+import { createWorkflowSchema } from '../../forms/workflow-form.schema';
+import { addWorkflow, updateWorkflow, fetchWorkflow, fetchWorkflows } from '../../redux/actions/workflow-actions';
+import SummaryContent from './summary_content';
+
+const AddWorkflowModal = ({
+  history: { goBack },
+  addWorkflow,
+  addNotification,
+  fetchWorkflow,
+  fetchWorkflows,
+  initialValues,
+  updateWorkflow,
+  initialGroups,
+  workflowId,
+  rbacGroups
+}) => {
+  useEffect(() => {
+    if (workflowId) {
```

Comment:
This hook supplements `componetDidMount`  method. So if there is no ID when component mounts the data is never fetched.

Either remove the condition or add the id to the list of changing props to trigger hook when the prop changes:
```javscript
useEffect(() => {
...
}, [ workflowId ])
```